src/mlpack/methods/lmnn/lmnn_main.cpp

```diff
@@ -344,10 +344,12 @@ static void mlpackMain()
     LMNN<LMetric<2>> lmnn(data, labels, k);
     lmnn.Regularization() = regularization;
     lmnn.Range() = range;
+    lmnn.LinearScan() = shuffle;
     lmnn.Optimizer().StepSize() = stepSize;
     lmnn.Optimizer().MaxIterations() = passes * data.n_cols;
     lmnn.Optimizer().Tolerance() = tolerance;
-    lmnn.Optimizer().Shuffle() = shuffle;
+    // Bypass LinearScan. It will be handle in shuffle().
+    lmnn.Optimizer().Shuffle() = true;
```

Comment:
I think it's `true` by default actually, so this line may not be necessary.  But it doesn't hurt anything.